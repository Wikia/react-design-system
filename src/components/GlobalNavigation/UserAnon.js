import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';
import linkType from './linkType';

const UserAnon = ({anonLinks}) => {
  if (!anonLinks) {
    return null;
  }

  return (
    <div className="wds-global-navigation__user-menu wds-has-shadow wds-dropdown">
      <div className="wds-dropdown__toggle">
        <div className="wds-avatar">
          <div className="wds-avatar__inner-border" />
          <svg className="wds-avatar__image">
            <use xlinkHref="#wds-avatar-icon-user" />
          </svg>
        </div>
        <svg className="wds-icon wds-icon-tiny wds-dropdown__toggle-chevron">
          <use xlinkHref="#wds-icons-dropdown-tiny" />
        </svg>
      </div>
      <div className="wds-global-navigation__dropdown-content wds-is-right-aligned wds-dropdown__content">
        <ul className="wds-list wds-has-lines-between">
          {anonLinks.map(item => (
            <li key={item.tracking_label}>
              {item.caption && (
                <div className="wds-global-navigation__user-menu-dropdown-caption">
                  {item.caption}
                </div>
              )}
              <Link
                className={`wds-is-full-width wds-button ${item.secondary && 'wds-is-secondary'}`}
                link={item}
              />
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

UserAnon.propTypes = {
  anonLinks: PropTypes.arrayOf(linkType),
};

UserAnon.defaultProps = {
  anonLinks: null,
};

export default UserAnon;
