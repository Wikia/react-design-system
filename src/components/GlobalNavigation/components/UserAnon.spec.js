import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import sinon from 'sinon';

import UserAnon from './UserAnon';

test('UserAnon shouldn\'t render with null', () => {
  const component = renderer.create(
    <UserAnon user={null} />
  );
  expect(component.toJSON()).toBe(null);
});

test('UserAnon should render correctly with data', () => {
  const anonLinks = [
    {
      type: 'link-full-button',
      href: 'https://www.sandbox-content.wikia.com/signin?redirect=<referrer>',
      title: 'Sign-in',
      tracking_label: 'account.sign-in',
    },
    {
      type: 'link-full-button-secondary',
      href: 'https://www.sandbox-content.wikia.com/register?redirect=<referrer>',
      title: 'Register',
      tracking_label: 'account.register',
      caption: 'Don\'t have an account?',
    },
  ];
  const component = renderer.create(
    <UserAnon anonLinks={anonLinks} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
