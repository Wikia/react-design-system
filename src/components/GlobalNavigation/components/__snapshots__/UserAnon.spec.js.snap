// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAnon should render correctly with data 1`] = `
<div
  className="wds-global-navigation__user-menu wds-has-shadow wds-dropdown"
>
  <div
    className="wds-dropdown__toggle"
  >
    <div
      className="wds-avatar"
    >
      <div
        className="wds-avatar__inner-border"
      />
      <svg
        className="wds-avatar__image"
      >
        <use
          xlinkHref="#wds-avatar-icon-user"
        />
      </svg>
    </div>
    <svg
      className="wds-icon wds-icon-tiny wds-dropdown__toggle-chevron"
    >
      <use
        xlinkHref="#wds-icons-dropdown-tiny"
      />
    </svg>
  </div>
  <div
    className="wds-global-navigation__dropdown-content wds-is-right-aligned wds-dropdown__content"
  >
    <ul
      className="wds-list wds-has-lines-between"
    >
      <li>
        <a
          className="wds-is-full-width wds-button false"
          data-tracking-label="account.sign-in"
          href="https://www.sandbox-content.wikia.com/signin?redirect=<referrer>"
        >
          Sign-in
        </a>
      </li>
      <li>
        <div
          className="wds-global-navigation__user-menu-dropdown-caption"
        >
          Don't have an account?
        </div>
        <a
          className="wds-is-full-width wds-button wds-is-secondary"
          data-tracking-label="account.register"
          href="https://www.sandbox-content.wikia.com/register?redirect=<referrer>"
        >
          Register
        </a>
      </li>
    </ul>
  </div>
</div>
`;
